### PR TITLE
Add install script to conditionally launch node-pre-gyp

### LINKED ISSUE #674: Errors running npm install on Windows, missing files?
Trying to use Realm in my React Native app on Windows 10. The output is pasted below. It seems like it's missing the realm pre built stuff for my node version?

Node: 6.9.1
npm: 3.10.8
(React Native: 0.37)

```
c:\code\ReduxNavigation>npm install realm --save

> realm@0.15.0 install c:\code\ReduxNavigation\node_modules\realm
> node-pre-gyp install --fallback-to-build

node-pre-gyp ERR! Tried to download(404): https://static.realm.io/node-pre-gyp/realm-v0.15.0-node-v48-win32-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for realm@0.15.0 and node@6.9.1 (node-v48 ABI) (falling back to source compile with node-gyp)

c:\code\ReduxNavigation\node_modules\realm>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" clean )  else (node "" clean )

c:\code\ReduxNavigation\node_modules\realm>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" configure --fallback-to-build --module=c:\code\ReduxNavigation\node_modules\realm\compiled\node-v48_win32_x64\realm.node --module_name=realm --module_path=c:\code\ReduxNavigation\node_modules\realm\compiled\node-v48_win32_x64 )  else (node "" configure --fallback-to-build --module=c:\code\ReduxNavigation\node_modules\realm\compiled\node-v48_win32_x64\realm.node --module_name=realm --module_path=c:\code\ReduxNavigation\node_modules\realm\compiled\node-v48_win32_x64 )
Traceback (most recent call last):
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\gyp_main.py", line 16, in <module>
    sys.exit(gyp.script_main())
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 545, in script_main
    return main(sys.argv[1:])
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 538, in main
    return gyp_main(args)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 523, in gyp_main
    generator.GenerateOutput(flat_list, targets, data, params)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 1988, in GenerateOutput
    generator_flags))
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 927, in _GenerateProject
    return _GenerateMSVSProject(project, options, version, generator_flags)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 1028, in _GenerateMSVSProject
    _AddAccumulatedActionsToMSVS(p, spec, actions_to_add)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 468, in _AddAccumulatedActionsToMSVS
    cmd=command)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 438, in _AddCustomBuildToolForMSVS
    _ConfigFullName(config_name, c_data), tools=[tool])
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\MSVSProject.py", line 181, in AddFileConfig
    raise ValueError('AddFileConfig: file "%s" not in project.' % path)
ValueError: AddFileConfig: file "..\..\binding.gyp" not in project.
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:305:16)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Windows_NT 10.0.10586
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-build" "--module=c:\\code\\ReduxNavigation\\node_modules\\realm\\compiled\\node-v48_win32_x64\\realm.node" "--module_name=realm" "--module_path=c:\\code\\ReduxNavigation\\node_modules\\realm\\compiled\\node-v48_win32_x64"
gyp ERR! cwd c:\code\ReduxNavigation\node_modules\realm
gyp ERR! node -v v6.9.1
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'node-gyp.cmd configure --fallback-to-build --module=c:\code\ReduxNavigation\node_modules\realm\compiled\node-v48_win32_x64\realm.node --module_name=realm --module_path=c:\code\ReduxNavigation\node_modules\realm\compiled\node-v48_win32_x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (c:\code\ReduxNavigation\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:106:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:877:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
node-pre-gyp ERR! System Windows_NT 10.0.10586
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "c:\\code\\ReduxNavigation\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd c:\code\ReduxNavigation\node_modules\realm
node-pre-gyp ERR! node -v v6.9.1
node-pre-gyp ERR! node-pre-gyp -v v0.6.31
node-pre-gyp ERR! not ok
Failed to execute 'node-gyp.cmd configure --fallback-to-build --module=c:\code\ReduxNavigation\node_modules\realm\compiled\node-v48_win32_x64\realm.node --module_name=realm --module_path=c:\code\ReduxNavigation\node_modules\realm\compiled\node-v48_win32_x64' (1)
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "realm" "--save"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE

npm ERR! realm@0.15.0 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the realm@0.15.0 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the realm package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs realm
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls realm
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     c:\code\ReduxNavigation\npm-debug.log
```

### PR DESCRIPTION
**Pull request by kristiandupont**
This fixes #674

### CODE REVIEW

**Comment by kristiandupont** [comment]
retest this please

**Comment by kristiandupont** [comment]
retest this please

**Comment by alazier** [comment]
This is great. Do we need to return the exit code for node-gyp in case of failure?

**Comment by kristiandupont** [comment]
Passing through the error code is a good idea. As for the `--build-from-source` parameter, this is only called by NPM during install and I don't think that we would ever call it manually as it's mostly in place to distinguish between RN and Node installs. But I can move it if you think it's important.

**Comment by kristiandupont** [comment]
retest this please

File: scripts/install.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+if (isNode) {
+  if (process.platform === 'win32') {
+    console.error('ERROR: Realm is not yet supported on Windows');
+    process.exit(-1);
+  }
+
+  // Execute "node-pre-gyp install --fallback-to-build
+  var pregyp = spawn('node-pre-gyp', ['install', '--fallback-to-build']);
+  pregyp.stdout.on('data', function (data) { console.log(data.toString()); });
```
Comment (fealebenpae) [inline scripts/install.js]: `pregyp.stdout.pipe(process.stdout)` might be cleaner/better here.

**Pull request merged**